internal/terraform/lang/parser.go

```diff
@@ -37,10 +37,19 @@ type parser struct {
 }
 
 func ParserSupportsTerraform(v string) error {
-	tfVersion, err := version.NewVersion(v)
+	rawVer, err := version.NewVersion(v)
 	if err != nil {
 		return err
 	}
+
+	// Assume that alpha/beta/rc prereleases have the same compatibility
+	segments := rawVer.Segments64()
+	segmentsOnly := fmt.Sprintf("%d.%d.%d", segments[0], segments[1], segments[2])
```

Comment:
Perhaps it would be useful to upstream these two lines into `go-version` as `Version.RawVersion()` or something like that. Not something I'd want to block this PR on, but thinking out loud.

Reply:
Yeah, I think we should just start implementing this in terraform-exec and see what feels right there. I'd rather not have to have a lot of this stuff in different packages we need to think/worry about and terraform-exec will handle schema dumping, we could also add lock file watching to it, etc.